tso-server/src/main/java/org/apache/omid/tso/PersistenceProcessorImpl.java

```diff
@@ -146,20 +146,30 @@ public void addCommitRetryToBatch(long startTimestamp, Channel c, MonitoringCont
     }
 
     @Override
-    public void addAbortToBatch(long startTimestamp, Channel c, MonitoringContext context)
+    public void addAbortToBatch(long startTimestamp, Channel c, MonitoringContext monCtx)
             throws Exception {
 
-        currentBatch.addAbort(startTimestamp, c, context);
+        currentBatch.addAbort(startTimestamp, c, monCtx);
         if (currentBatch.isFull()) {
             triggerCurrentBatchFlush();
         }
 
     }
 
     @Override
-    public void addTimestampToBatch(long startTimestamp, Channel c, MonitoringContext context) throws Exception {
+    public void addTimestampToBatch(long startTimestamp, Channel c, MonitoringContext monCtx) throws Exception {
 
-        currentBatch.addTimestamp(startTimestamp, c, context);
+        currentBatch.addTimestamp(startTimestamp, c, monCtx);
+        if (currentBatch.isFull()) {
```

Comment:
Every function that adds an item to a batch should check whether the batch is full and trigger a flush in case it is. You can take a look at add{Timestamp, Abort, Commit, CommitRetry}ToBatch functions. We can wrap this check in some epilog.